src/hooks/useDPI.ts

```diff
@@ -43,21 +14,9 @@ export default function useDPI() {
   );
 
   const fetchDPIData = useCallback(async () => {
-    const adapterRegistry = new Contract(
-      DEFI_SDK_ADAPTER_REGISTRY_ADDRESS,
-      defiSdkAdapterRegistryABI,
-      web3Provider!
-    );
-    const result = await adapterRegistry.getFinalFullTokenBalance(
-      TOKENSETS_V2,
-      DPI_ADDRESS
-    );
-    const defiPulseData = {
-      base: getTokenData(result.base),
-      underlying: map(result.underlying, token => getTokenData(token)),
-    };
+    const defiPulseData = await getDPIBalance();
```

Comment:
`defiPulseData` could potentially be undefined

Reply:
good catch, added optional chaining below